### PR TITLE
ref(replay): replace replays/contextIcon import with PlatformIcon

### LINKED ISSUE #73945: [Replay] Replace references to ContextIcon imports with equivalent PlatformIcon
Check the replay list page, as well as replay details page.

Specific files to look at:
https://github.com/getsentry/sentry/blob/master/static/app/components/replays/contextIcon.tsx#L20
https://github.com/getsentry/sentry/blob/master/static/app/views/replays/replayTable/tableCell.tsx#L10
https://github.com/getsentry/sentry/blob/master/static/app/views/replays/detail/browserOSIcons.tsx#L4

### PR DESCRIPTION
**Pull request by aliu39**
Fixes https://github.com/getsentry/sentry/issues/73945

Bumping to the latest version of [the platformicons package](https://www.npmjs.com/package/platformicons?activeTab=code), since the current one doesn't have Chrome, Safari, and maybe some others.

I kinda eye-balled the size, when using 14 the new icons were clearly smaller than the old.
ContextIcon, 14px:
Screenshot 2024-07-12 at 10 29 24 AM
PlatformIcon (NEW), 18px: Didn't translate the name for edge here, but it should work in the preview deployment
Screenshot 2024-07-12 at 10 29 49 AM

TODO:
- [x] translate "mac-os-x" to "apple", and check other names.
    - [x] can maybe write a new function to replace `events/contexts/utils/generateIconName`
- [ ] use lazy loading like before. Is this necessary? I was getting type errors in the lazy() statement earlier

### CODE REVIEW

**Comment by Jesse-Box** [comment]
Can comment when I see pics or a deploy I can check out @aliu39 :)

File: static/app/components/replays/contextIcon.tsx
Comment (michellewzhang) [inline static/app/components/replays/contextIcon.tsx:18]:
```suggestion
// theme.iconSizes.sm + 4, to make up for padding
```
conventionally we write this

File: static/app/components/replays/contextIcon.tsx
Comment (michellewzhang) [inline static/app/components/replays/contextIcon.tsx:19]: i believe `theme.iconSizes.md` is equivalent to `18px`, can we reuse theme here?

File: static/app/utils/replays/icons.tsx
Comment (ryan953) [inline static/app/utils/replays/icons.tsx:28]:
```suggestion
  name: string,
  version: string | undefined
```

File: static/app/utils/replays/icons.tsx
Comment (ryan953) [inline static/app/utils/replays/icons.tsx:1]: Lets name this file `generatePlatformIconName.tsx` because that is all that is being exported. no need to hide it

File: static/app/utils/replays/icons.tsx
Comment (michellewzhang) [inline static/app/utils/replays/icons.tsx:36]:
```suggestion
  const contextName = generateIconName(name, version);

  const platformName = contextName in PLATFORM_ALIASES ? PLATFORM_ALIASES[contextName] : platformName;
 
```
can do something like this (with better formatting) -- it's called a ternary!

**Review by ryan953 — APPROVED** [review]

File: static/app/utils/replays/icons.tsx
Comment (michellewzhang) [inline static/app/utils/replays/icons.tsx:20]: how did you test which platforms needed aliases??

Comment (aliu39) [reply static/app/utils/replays/icons.tsx:20]: It was manual - I used ctrl-F to check the contextIcon stories vs the svg/ folder in the platformicons repo

Comment (michellewzhang) [reply static/app/utils/replays/icons.tsx:20]: not the manual labor✊

Comment (aliu39) [reply static/app/utils/replays/icons.tsx:20]: not the not the

**Comment by aliu39** [comment]
Screenshot 2024-07-12 at 12 12 58 PM
The edge-legacy icon that wasn't working in the ss above.

Gonna wait for @Jesse-Box approval to merge!

**Review by michellewzhang — APPROVED** [review]

**Comment by Jesse-Box** [comment]
> @aliu39 can we make the icon 16px in size, but give the icon itself a 1px border using `translucentGray100`?

@ryan953 After we're happy with a solution, I sense, we'll need to “lift up” our solution so that other teams can benefit

**Comment by aliu39** [comment]
> @aliu39 can we make the icon 16px in size, but give the icon itself a 1px border using `translucentGray100`?

I think this would make the icon smaller than what we currently have. The images imported from platformicons all seem to have built-in translucent padding. Not sure if we can change that package or not

**Comment by ryan953** [comment]
> ## [Bundle](https://app.codecov.io/gh/getsentry/sentry/pull/74173?dropdown=bundle) Report
> Changes will increase total bundle size by 339.94kB ⬆️
> 
> Bundle name	Size	Change
> app-webpack-bundle-array-push	28.28MB	339.94kB ⬆️

@nicholas-codecov is the bundle size check targetting the PR base? If so this is a big bundle change, we might want the lazy loading back to reduce the numbers again.

**Comment by aliu39** [comment]
@ryan953 Drafted lazy load at https://github.com/getsentry/sentry/pull/74284

**Comment by ryan953** [comment]
@aliu39 since the codecov metric might be off still, lets defer the lazy loading stuff because it's just adding complexity anyway.

**Comment by Jesse-Box** [comment]
> > @aliu39 can we make the icon 16px in size, but give the icon itself a 1px border using `translucentGray100`?
> 
> I think this would make the icon smaller than what we currently have. The images imported from platformicons all seem to have built-in translucent padding. Not sure if we can change that package or not

Yeah, my goals aren't to have them at the same size. 16px is pretty standard size for our system icons, and I want align to those values.

Just so you know, all the platform icons come with a background color so that light and dark themes look equally as good.

**Comment by aliu39** [comment]
@Jesse-Box gotcha! I'll make the change

**Comment by aliu39** [comment]
Like this? @Jesse-Box 
Screenshot 2024-07-17 at 3 02 25 PM
Screenshot 2024-07-17 at 3 02 57 PM

**Comment by Jesse-Box** [comment]
> Like this? @Jesse-Box
> 
> After	Before
> Screenshot 2024-07-17 at 3 02 25 PM	Screenshot 2024-07-17 at 3 07 19 PM
> Screenshot 2024-07-17 at 3 02 57 PM	Screenshot 2024-07-17 at 3 08 11 PM

🚢 it

**Review by Jesse-Box — APPROVED** [review]

**Pull request merged**
